data/types.go

```diff
@@ -3,86 +3,193 @@ package data
 import (
 	"errors"
 	"fmt"
+	"strings"
 )
 
 var (
 	// ErrUnknownType is returned if the type cannot be determined.
 	ErrUnknownType = errors.New("data: unknown type")
 )
 
-// Type is a known type of file such as JPEG or PNG. Only known types of files
-// are copied from source to destination.
+// Type definitions.
+const (
+	// UnknownType is the zero value for Type, meaning it is unknown.
+	UnknownType Type = ""
+
+	// Image formats.
+	JPG = "jpg" // Standard JPG file.
+	PNG = "png" // Standard PNG file.
+	GIF = "gif" // Standard GIF file.
+)
+
+// Encoding definitions.
+const (
+	// Native is the zero value for Encoding, meaning no encoding.
+	Native Encoding = ""
+
+	Tar  = "tar"
+	GZip = "gz"
+)
+
+// Class definitions.
+const (
+	Unclassified Class = ""
+	Image              = "image"
+)
+
+// Type is a known kind of file such as JPEG or PNG.
 type Type string
 
 func (t Type) String() string {
+	if t == "" {
+		return "data:unknown"
+	}
 	return string(t)
 }
 
-// Ext returns the type's file extension.
+// Ext returns the type's standard file extension.
 func (t Type) Ext() string {
-	enc := t.Enc()
-	if enc == Native {
-		return string(t)
+	return string(t)
+}
+
+// Class returns the type's class: image, catalog, etc.
+func (t Type) Class() Class {
+	return classOfType[t]
+}
+
+// Ok return true if the type is defined.
+func (t Type) Ok() bool {
+	_, ok := types[t]
+	return ok
+}
+
+// ParseExt parses an extension, returning the Stored format.
+func ParseExt(str string) (Stored, error) {
+	if str == "" {
+		return Stored{}, nil
+	}
+	parts := strings.Split(str, ".")
```

Comment:
No because that function finds the "one dot" extension from any string, but the job here is to parse a "multiple dots" extension itself into its parts.